File: ClientApp/src/app/interfaces/product/response/warrantyResponse.ts
import { ByBaseResponse } from "../../by-base-response";
import { ProductsListResponse } from "./products-list-response";

export interface WarrantyResponse extends ByBaseResponse {
	id: string;
	name: string;
	type: string;
	status: string;
	period: number;
	description: string;
	conditions: string;
	products: ProductsListResponse[];
}

File: ClientApp/src/app/interfaces/Product/Response/WarrantyResponse.cs
namespace Contracts.Model.Product.Response;

public class WarrantyResponse : ByBaseResponse
{
	public Guid Id { get; set; }
	public string? Name { get; set; }
	public int Type { get; set; }
	public string Status { get; set; }
	public int Period { get; set; }
	public string Description { get; set; }
	public string Conditions { get; set; }
	public IEnumerable<ProductsListResponse> Products { get; set; }
}